test/amoc_api_status_handler_SUITE.erl

```diff
@@ -4,260 +4,76 @@
     Config.
 
-end_per_testcase(test_load_good, Config) ->
-    file:delete("scenarios/mongoose_simple.erl"),
-    file:del_dir("scenarios"),
-    file:del_dir("ebin"),
-    Config;
-
-end_per_testcase(test_load_compilation_err, Config) ->
-    file:del_dir("scenarios"),
-    Config;
-
-end_per_testcase(test_start_success, Config) ->
-    file:delete("scenarios/mongoose_simple.erl"),
-    file:del_dir("scenarios"),
-    file:delete("ebin/mongoose_simple.beam"),
-    file:del_dir("ebin"),
-    code:purge(mongoose_simple),
-    Config;
-
 end_per_testcase(_, _Config) ->
     application:stop(inets),
     application:stop(amoc),
     amoc_api:stop().
 
-test_success(_Config) ->
+works_when_amoc_running_offline(_Config) ->
     %% given
     given_applications_started(),
     %% when
-    Result = httpc:request("http://localhost:4000/status"),
-    {ok, {{_HttpVsn, CodeHttp, _Status}, _, _}} = Result,
-    %% then
-    ?assertEqual(200, CodeHttp).
-
-test_fail(_Config) ->
-    %% given
-    given_http_api_started(),
-    %% when
-    Result = httpc:request("http://localhost:4000/status"),
-    {ok, {{_HttpVsn, CodeHttp, _Status}, _, _}} = Result, 
+    Status = amoc_api_status_handler:get_status(),
     %% then
-    ?assertEqual(503, CodeHttp).
-
+    ?assert(Status).
 
-test_list(_Config) ->
-    %% given
-    given_applications_started(),
-    meck:new(file, [unstick, passthrough]),
-    Fun = fun(_) -> {ok, ["scenario.erl"]} end,
-    meck:expect(file, list_dir, Fun),
-    %% when    
-    {CodeHttp, JSON} = amoc_api_scenario_handler:process_json(dd,#state{action = list}),
-    meck:unload(file),
-    [ {AtomB, FilenamesB} ] = jsx:decode(JSON),
-    %% then
-    ?assertEqual(200, CodeHttp),
-    ?assertEqual(<<"scenarios">>, AtomB),
-    ?assertEqual(true, is_list(FilenamesB)).
-        
-test_start_fail_no_scenario(_Config) ->
-    %% given
-    given_applications_started(),
-    Term = [ 
-            {<<"scenario">>, <<"no_exist">>},
-            {<<"users">>, <<"10">>}
-          ],
-    %% when
-    {CodeHttp, JSON} = amoc_api_scenario_handler:process_json(Term,#state{action = start}),
-    %% then
-    ?assertEqual(500, CodeHttp),
-    ?assertEqual(
-                [{<<"error">>, <<"module_not_exists">>}],
-                jsx:decode(JSON)).
-                  
-test_start_success(_Config) ->
+works_when_amoc_running_online(_Config) ->
     %% given
     given_applications_started(),
-    Term = [ 
-            {<<"scenario">>, <<"mongoose_simple">>},
-            {<<"users">>, 10}
-           ],
-    meck:new(amoc_dist, [unstick]),
-    Fun = fun(_,1,_) -> ok end,
-    meck:expect(amoc_dist, do, Fun),
     %% when
-    Result = amoc_api_scenario_handler:process_json(Term,#state{action = start}),
-    meck:wait(amoc_dist, do, ['mongoose_simple', 1, 10], 1000),
-    meck:unload(amoc_dist),
-    {CodeHttp,JSON} = Result,
+    Result = httpc:request(get, {get_url() ++  "/status", [{"Accept", "application/json"}]}, [], []),
```

Comment:
Can I add that you should not use `httpc`, in general? For tests, it's ok, but since you're using this project to learn, I think you can take this opportunity to start using [fusco](https://github.com/esl/fusco) as an http client.